src/adapters/MeetingsAdapter.js

```diff
@@ -0,0 +1,90 @@
+import {throwError} from 'rxjs';
+
+import WebexAdapter from './WebexAdapter';
+
+/**
+ * This is a base class that defines the interface that maps meetings data.
+ * Developers that want to extend `MeetingsAdapter` must implement all of its methods,
+ * adhering to the exact parameters and structure of the returned objects.
+ */
+export default class MeetingsAdapter extends WebexAdapter {
+  /**
+   * A meeting object that allows users to have a WebRTC meeting.
+   *
+   * @typedef {Object} Meeting
+   * @property {string}  ID The meeting identifier.
+   * @property {string}  meetingTitle The title of the meeting.
+   * @property {Date}    meetingTimeStart The time and date of the start of the meeting.
+   * @property {Date}    meetingTimeEnd The time and date of the end of the meeting.
+   * @property {Boolean} localAudioMuted If the local user has muted their audio.
+   * @property {Boolean} localVideoMuted If the local user has muted their video.
+   * @property {Boolean} meetingJoined If the local user has joined the meeting.
+   */
+
+  /**
+   * Returns an observable that emits meeting data of the given ID.
+   *
+   * @param {string} ID  ID of the meeting to get.
+   * @returns {Observable.<Meeting>}
+   * @memberof MeetingsAdapter
+   */
+  // eslint-disable-next-line no-unused-vars
+  getMeeting(ID) {
+    return throwError(new Error('getMeeting(ID) must be defined in MeetingsAdapter'));
+  }
+
+  /**
+   * Returns a promise that resolved with the meeting ID that matches
+   * the requested parameters.
+   *
+   * @param {string} destinationID
+   * @param {string} destinationType
+   * @returns {Promise.<string>}
+   * @memberof MeetingsAdapter
+   */
+  // eslint-disable-next-line no-unused-vars
+  getMeetingID(destinationID, destinationType) {
```

Comment:
I would rather this return an observable that completes after sending the meeting ID. For consistency with other asynchronous-type methods, as well as composability with other observable streams.

Reply:
Sure